src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt

```diff
@@ -264,31 +263,62 @@ class RealHyperShard(
         checkIsKotlin(file)
 
         val tests = ArrayList<String>()
-        val parser = Parser()
-        val code = String(Files.readAllBytes(file.toPath()))
-        val ktFile = parser.parsePsiFile(code)
-        val allDeclarations = Arrays.stream(ktFile.children).toList()
-        for (declaration in allDeclarations) {
-            val ktClass = declaration as? KtClass ?: continue
+        val source = AstSource.File(file.toPath().toString())
+        val parseResult = KotlinGrammarAntlrKotlinParser.parseKotlinFile(source)
+        val allDeclarations = parseResult.summary(attachRawAst = false).get()
+
+        // Find header
+        val classIdentifier = fun (): String {
+            for (declaration in allDeclarations) {
+                val packageHeader = declaration as? PackageHeader ?: continue
+                return packageHeader.identifier.joinToString(separator = ".") { it.identifier }
+            }
+            throw Exception("Expected a PackageHeader attribute in source.")
+        }()
+
+        val ktClasses: List<KlassDeclaration> = allDeclarations.mapNotNull {
+            if (it is KlassDeclaration && it.keyword == "class") {
+                it
+            } else { null }
+        }
+
+        for (ktClass in ktClasses) {
+            val shouldProcessTestMethods = shouldProcessTestMethods(
+                    ktClass,
+                    annotationValue,
+                    notAnnotationValue
+            )
 
-            val shouldProcessTestMethods = shouldProcessTestMethods(ktClass, annotationValue,
-                notAnnotationValue)
             if (!shouldProcessTestMethods) {
                 continue
             }
 
-            for (element in declaration.children) {
-                val ktClassBody = element as? KtClassBody ?: continue
+            for (element in ktClass.children) {
+                if ((element as? DefaultAstNode)?.description != "classBody") {
```

Comment:
nit, this reads a bit dense, sanity check if element is not default as node do you want this to evaluate to true?